### ISSUE TITLE
Fix request: Kotlin AvoidMultipleConcatStatements false positive on MutableList

### ISSUE THREAD

**Issue by stokpop** [jborgers/PMD-jPinpoint-rules#651]
False positive, using `+=` on `MutableList`, so not on `String` as the rule assumes:

```kotlin
class Foo(private val isActive: Boolean, private val canCreate: Boolean) {

    private fun calculate(myList: List<String>): List<String> {

        val mutableList = myList.toMutableList()

        if (isActive) {
            mutableList += "one" // false positive, not two lines with += are needed to trigger this rule
            if (canCreate) {
                mutableList += listOf("two", "three") // false positive
            }
        } else {
            mutableList.removeAll(listOf("one", "two"))
        }
        return myList
    }
}
```

There are probably other mutable sets and maps that also gives false positives in the same way as the mutable list.

**Comment by stokpop** [comment]
Also for this (no false positive when the String.join is removed):

```kotlin
class Bar (
    val one: String,
    val two: String,
    val three: String,
) {

    override fun toString(): String = java.lang.String.join("-", one, two, three)

    override fun hashCode(): Int {
        var result = one.hashCode()
        result = 31 * result + two.hashCode()
        result = 31 * result + three.hashCode() // <-- false positive
        return result
    }
}
```
